src/test/java/cloud/fogbow/common/plugins/cloudidp/openstack/v3/OpenStackIdentityProviderPluginTest.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    // test case: When invoking the getCloudUser method with a null credential map,
+    // it must throw an UnexpectedException.
+    @Test(expected = NullPointerException.class) // verify
+    public void testGetCloudUserWithNullCredentialsMap() throws FogbowException {
+        // set up
+        Map<String, String> credentials = null;
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with an empty credential
+    // map, it must throw an UnexpectedException.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testGetCloudUserWithEmptyCredentialsMap() throws FogbowException {
+        // set up
+        Map<String, String> credentials = new HashMap<String, String>();
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // without a user name, it must throw an UnexpectedException.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testGetCloudUserWithoutUsername() throws FogbowException {
+        // set up
+        Map<String, String> credentials = createCredentials(null, ANY_VALUE, ANY_VALUE, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // without a password, it must throw an UnexpectedException.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testGetCloudUserWithoutPassword() throws FogbowException {
+        // set up
+        Map<String, String> credentials = createCredentials(ANY_VALUE, null, ANY_VALUE, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // without a domain, it must throw an UnexpectedException.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testGetCloudUserWithoutDomain() throws FogbowException {
+        // set up
+        Map<String, String> credentials = createCredentials(ANY_VALUE, ANY_VALUE, null, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // without a domain, it must throw an UnexpectedException.
+    @Test(expected = UnexpectedException.class) // verify
+    public void testGetCloudUserWithoutProject() throws FogbowException {
+        // set up
+        Map<String, String> credentials = createCredentials(ANY_VALUE, ANY_VALUE, ANY_VALUE);
+
+        // exercise
+        this.plugin.getCloudUser(credentials);
+    }
+
+    // test case: When invoking the getCloudUser method with a credential map
+    // with correct data, without providing projectId
```

Comment:
this descriptions misses the expected condition. Note that, it requires this pattern "When X then Y"

Reply:
Done.